atlp-rwanda/atlp-devpulse-bn
# #151 Applicant be able to apply job

### PR Description
This PR is for applicant be able to apply job post

### Description of tasks that were expected to be completed

- Applicant be able to apply job post
- Applicant be able to fill the google form for the job posted


### How has this been tested?
- Clone repo
- Checkout branch `ft-apply-job`
- Make sure that you set the environment variables referring to `.env.example`
- Run `npm install`, then `npm run dev` to check if the app is running successfully
- Navigate to `localhost:4000`

- `mutation SaveSheetData` passing `google sheet link` and `range` sheet name `ex`: `sheet1`,`form response 1` since it for google form by default the `range ` is `Form Response 1`
```
{  "sheetLink":"https://docs.google.com/spreadsheets/d/1Umv-0JqnqDMw586PJK41-ve2VZ-zaQ7J_EkynE6cdnE/edit?usp=sharing",
  "range": "Form Responses 1"
}
```
- Use query `GetAllAppliedJobs` and query `GetAppliedJob` to get the applied user's data

### Steps to Create an API Key:
1. Go to the `Google Cloud Console`.
2. Select your project (or create a new one).
3. Navigate to `APIs & Services` > `Credentials.`
4. Click Create `Credentials and select API Key.`
5. Copy the generated `API key`.
6. paste API Key in your environment variable to `API_KEYS`

### Track PR (issue number & link)
The issue link is the on front end click [here](https://github.com/atlp-rwanda/atlp-devpulse-fn/issues/151)

### Screenshots (If appropriate especially on the frontend)
N/A
